src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,299 @@
+            schoolQuantity = 20;
+            minDistToMario = 1500.0f;
+            fishAnimation = blue_fish_seg3_anims_0301C2B0;
             break;
+            
+        // Blue fish with a quanitiy of five.
         case 1:
-            model = MODEL_FISH;            sp38 = 5;            sp30 = 1500.0f;            sp2C = blue_fish_seg3_anims_0301C2B0;
+            model = MODEL_FISH;
+            schoolQuantity = 5;
+            minDistToMario = 1500.0f;
+            fishAnimation = blue_fish_seg3_anims_0301C2B0;
             break;
+            
+        // Cyan fish with a quanitiy of twenty.
         case 2:
-            model = MODEL_CYAN_FISH;            sp38 = 20;            sp30 = 1500.0f;            sp2C = cyan_fish_seg6_anims_0600E264;
+            model = MODEL_CYAN_FISH;
+            schoolQuantity = 20;
+            minDistToMario = 1500.0f;
+            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
+            
+        // Cyan fish with a quanitiy of five.
         case 3:
-            model = MODEL_CYAN_FISH;            sp38 = 5;            sp30 = 1500.0f;            sp2C = cyan_fish_seg6_anims_0600E264;
+            model = MODEL_CYAN_FISH;
+            schoolQuantity = 5;
+            minDistToMario = 1500.0f;
+            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
     }
-    if (o->oDistanceToMario < sp30 || gCurrLevelNum == LEVEL_SA) {
-        for (i = 0; i < sp38; i++) {
-            sp28 = spawn_object(o, model, bhvFishGroup2);
-            sp28->oBehParams2ndByte = o->oBehParams2ndByte;
-            func_8029EE20(sp28, sp2C, 0);
-            translate_object_xyz_random(sp28, 700.0f);
+    /**
+     * If Distance to Mario is lower than 1500 or the current level is Secret Aquarium,
+     * then spawn and animate the schoolQuantity of fish.
+     * Fish moves at random with a max-range of 700.0f.
+     */
+    if (o->oDistanceToMario < minDistToMario || gCurrLevelNum == 20) {
+        for (i = 0; i < schoolQuantity; i++) {
+            fishObject = spawn_object(o, model, bhvFishGroup2);
+            fishObject->oBehParams2ndByte = o->oBehParams2ndByte;
+            func_8029EE20(fishObject, fishAnimation, 0);
+            translate_object_xyz_random(fishObject, 700.0f);
         }
-        o->oAction = 1;
+        o->oAction = FISH_ACTIVE;
     }
 }
 
-void ActionFish1(void) {
-    if (gCurrLevelNum != LEVEL_SA)
-        if (gMarioObject->oPosY - o->oPosY > 2000.0f)
-            o->oAction = 2;
+/**
+ * If the current level is not Secret Aquarium and the distance from
+ * mario->PosY minus o->oPosY is greater than 2000.0f then set oAction to FISH_RESPAWN.
+ */
+void fish_act_check_distance(void) {
+    if (gCurrLevelNum != 20) {
+        if (gMarioObject->oPosY - o->oPosY > 2000.0f) {
+            o->oAction = FISH_RESPAWN;
+        }
+    }
 }
 
-void ActionFish2(void) {
-    o->oAction = 0;
+/**
+ * set oAction flag to FISH_INIT.
+ */
+void fish_act_set(void) {
```

Comment:
I don't like the names `fish_act_check_distance` or `fish_act_set`. I think the names you used in the action macros are much better.